### PR TITLE
Email verification

### PR DESCRIPTION
**Pull request by zwolf**
Adds the Confirmable module for Devise along with the requisite database attributes. A "please confirm your email address" email is now sent to the user on account creation. That email is using the default template ([located here](https://github.com/zooniverse/panoptes/blob/email-verification/app/views/devise/mailer/confirmation_instructions.html.erb#L6)) and can be updated if desired. 

When a user creates a new account, the front end should inform them that they will receive a confirmation email (this currently happens on Panoptes' own reg page, but no one uses it). The link in that email will point to Panoptes directly so that the API can process the user's confirmation token. That "thank you for confirming" event will likely have to be updated to redirect to wherever makes the most sense since that form currently just refreshes the [panoptes API home page](https://panoptes.zooniverse.org/). Right now, successful I am redirecting successful confirmations to the zooniverse.org (production) home page. Is there a way to ensure the registration modal opens on PFE? I can include a query param that can trigger a "thank you for confirming your email address!" welcome flash message. Is that worth integrating into PFE, or should that wait until FEM is at the site root? **This is a todo and should be figured out & updated before this deploys to production.**

Also, when the user is confirmed they will then receive the standard Zooniverse welcome email. For reference, that template is [located here](https://github.com/zooniverse/panoptes/blob/email-verification/app/views/user_welcome_mailer/welcome_user.html.erb#L111) and hasn't needed to be updated in a while. 

Finally: there's a rake task to backfill all existing users with a `confirmed_at` value. This field is going to be used by Talk in order to prevent unconfirmed users from interacting with Talk beyond viewing public posts. This PR must be merged and this backfill run before the Talk PR merges.

This is gonna need a bit of hands on testing on staging, so once this and the Talk PR merge, that'll be the focus till production can be deployed safely.  

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/controllers/confirmations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ConfirmationsController < Devise::ConfirmationsController
+  protected
+
+  def after_confirmation_path_for(resource_name, resource)
```
[the remark this answers is not in the document]
Comment (zwolf) [reply app/controllers/confirmations_controller.rb:6]: Devise method override

File: spec/controllers/registrations_controller_spec.rb [hunk 2]
```diff
@@ -313,8 +313,8 @@
           expect(User.where(login: user_attributes[:login])).to_not exist
         end
 
-        it "should not queue a welcome worker to send an email" do
-          expect(UserWelcomeMailerWorker).to_not receive(:perform_async)
+        it "should not send a confirmation instructions email" do
+          expect_any_instance_of(User).to_not receive(:send_confirmation_instructions)
```
[the remark this answers is not in the document]
Comment (zwolf) [reply spec/controllers/registrations_controller_spec.rb:317]: Probably, but this is how these things are done around here.

**Comment by zwolf** [comment]
I apologize for the Hound, but these are some deep pages that are written in the olde ways. Didn't wanna rock the boat too much and make this even more annoying to review. Most all of it is syntax or "probably don't write specs this way" when all the other specs are written that way. Please lmk if I missed anything significant, though.

**Review by yuenmichelle1 — APPROVED** [review]
Looks fine to me. 
Some questions that you might know the answer to or answers come up through testing:
- If a user loses their confirmation email, (eg. never confirmed after a long time and now cannot post to talk because they are unconfirmed or another eg. If a person changes their email because they an created account on an email address they lost access to), is there a way to resend confirmation email?
- How long are `confirmation_token`s good for? ^ related to above

**Comment by zwolf** [comment]
> If a user loses their confirmation email, (eg. never confirmed after a long time and now cannot post to talk because they are unconfirmed or another eg. 

 There's a form at panoptes.zooniverse.org/users/confirmation, but no one uses those. One of the front end UX things we might need is a form/button pointed at the appropriate devise route in the user settings area to resend the email.

> If a person changes their email because they an created account on an email address they lost access to), is there a way to resend confirmation email?
If they confirmed with us once, that's good enough for me. No re-confirmation necessary.

> How long are confirmation_tokens good for? ^ related to above
Forever, they don't appear to change or expire.

**Pull request merged**